cudapoa/src/utils.cu

```diff
@@ -0,0 +1,159 @@
+/*
```

Comment:
can this file be called `utils.cpp`? or does it need to be compiled with `nvcc`?

Reply:
no, initially it was .cpp that I was getting some unrelated compilation errors. Because it includes `allocate_block.hpp` needs `nvcc`